app/equipment/armor.py

```diff
@@ -4,21 +4,15 @@
 
 
 class Armor(Equipment):
-    def __init__(self, part: str, power: int) -> None:
+    def __init__(self, part: str, protection: int) -> None:
         super().__init__(part)
-        self.power = power
+        self.power = protection
 
     @classmethod
     def apply(cls, knight: Any, source: dict) -> None:
-        armor_data = (
-            source.get(knight.name.lower(),
-                       source.get(knight.name.replace(" ", "_").lower(), {})
-                       ).get("armour", [])
-        )
-
-        if armor_data:
+        if source.get("armour"):
             knight.equip_list = []

```

Comment:
Btw, why to not set equip_list = [] in Knight.__init__ instead of this place? It seems to me that you don't use some checks for None of this attribute